### PR TITLE
Add working reload command

### PR DESCRIPTION
**Pull request by Kordyjan**
This PR fixes FOUR subtle bugs that worked in perfect coordination to leave the UI in a completely broken state silently after running the restart agent command. Also, now the command is accessible only via Eclipse's command palette.

## Test plan
- open chat view
- ask some question
- press `Cmd + 3`
- start typing `Restart Cody Agent`
- once the command shows up, press Enter

After a few seconds, Chat View should refresh and still work correctly.

## Future improvement
There should be some indication that Cody is reloading. We can show the spinner or fade the review.

### CODE REVIEW

File: plugins/cody-chat/src/com/sourcegraph/cody/chat/agent/ReloadHandler.java
Comment (olafurpg) [inline plugins/cody-chat/src/com/sourcegraph/cody/chat/agent/ReloadHandler.java]: Delete

File: plugins/cody-chat/src/com/sourcegraph/cody/workspace/WorkbenchListener.java
Comment (Kordyjan) [inline plugins/cody-chat/src/com/sourcegraph/cody/workspace/WorkbenchListener.java:93]: This works slightly differently than `asyncExec`, which adds runnable to the queue to be run on the UI thread once it is free.

`execute`, on the other hand, runs the task immediately if it is already on the UI thread. If called from another thread, it falls back to `asyncExec`.

It is used when the code is already expected to be called from the UI thread.

**Review by olafurpg — APPROVED** [review]
This was tricky indeed. Nice work! I have not tested this locally but overall LGTM. I'll leave it to @jamesmcnamara to test locally and merge

File: plugins/cody-chat/src/com/sourcegraph/cody/chat/agent/ReloadHandler.java
Comment (Kordyjan) [reply plugins/cody-chat/src/com/sourcegraph/cody/chat/agent/ReloadHandler.java]: Ooops

**Comment by jamesmcnamara** [comment]
So out of curiosity, what was causing the bug? I see a bunch of calls moved onto the UI thread and ensuring that we use the current version of the agent. Was it stale references?

**Comment by jamesmcnamara** [comment]
Ah I just saw the thread in slack. So yes, stale references to agent.

**Review by jamesmcnamara — APPROVED** [review]
Tested this on both Mac and Windows and it works fine. I agree that it is non-obvious that the window cannot be interacted with in during restart. Something to add.

**Comment by Kordyjan** [comment]
Stale references were one issue. When that was fixed, I observed that `connectWebviewToBrowser` was never called. That was caused by the `browserField.refresh()` call from the worker thread. This error was caught and would be logged, but the error handling code first tries to dispose of the agent. That caused another exception, again caused by illegal thread access. The code that caused it was outside of any try, so it just crashed a worker thread.

I was surprised that the second exception was a `NullPointerException`. This was because `getActiveWorkbenchWindow()`returns null when accessed from a worker thread. While it is documented, it is still weird and inconsistent with the rest of Eclipse API.

**Pull request merged**
